modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1886,6 +1886,17 @@ private void _validateEdge(
 				"System object definitions cannot inherit configurations",
 				"system-object-definitions-cannot-inherit-configurations");
 		}
+		else if ((objectDefinition1.isModifiableAndSystem() &&
+				  !objectDefinition2.isModifiableAndSystem()) ||
+				 (!objectDefinition1.isModifiableAndSystem() &&
+				  objectDefinition2.isModifiableAndSystem())) {
+
+			throw new ObjectRelationshipEdgeException(
+				"Inheritance between modifiable system and custom object " +
+					"definitions is not allowed",
+				"inheritance-between-modifiable-system-and-custom-object-" +
+					"definitions-is-not-allowed");
+		}
```

Comment:
I am uncertain whether the existing 'if' statement can be used to simplify my logic or if it is necessary to always sort alphabetically by error message

Reply:
:thinking: